bundle/edu.gemini.pot/src/test/scala/edu/gemini/spModel/target/env/Arbitraries.scala

```diff
@@ -309,4 +312,11 @@ trait Arbitraries extends edu.gemini.spModel.core.Arbitraries {
       arbitrary[Asterism.Single],
       arbitrary[GhostAsterism]
     ))
+
+  // Specific target environment types to avoid Scalacheck giving up.
+  // These can't go in GhostGens or we get a stack overflow error.
+  val genStandardResAsterismTargetEnvironment: Gen[TargetEnvironment] =
```

Comment:
If you make these return the subtype `GhostAsterism.StandardResolution`, etc. then you won't need the casts in the tests above. You can widen these via `.map[TargetEnvironment](identity)` in cases where you need the wider type.

Reply:
This one I'm a little confused about and not sure if I'm misunderstanding. You're referring to the tests in `AsterismConverterSpec` and the casts I'm doing there, right?

I want `TargetEnvironment`, preferably populated with an `Asterism` subclass of the correct type, a `GuideEnvironment` and possibly some `UserTarget`s, and `GhostAsterism.StandardResolution` is a subclass of `Asterism`... so I'm not sure how I can go from a `Gen[GhostAsterism.StandardResolution]` or an `Arbitrary[GhostAsterism.StandardResolution]` to a full-fledged `TargetEnvironment` without doing something like I'm doing here.

Wouldn't the widening require `GhostAsterism.StandardResolution` to be a subclass of `TargetEnvironment`?

The casting is making me feel dirty, though, so if I can eliminate it, that would be excellent.

(Sorry if I'm being a bit dense here, and I definitely appreciate the help.)

Reply:
Oh right, I'm sorry, I misread your code. Yeah I think that's probably the best you can do with the way things are structured now. Nevermind.